### PR TITLE
[CI] move e2e tests execution environment to kind

### PR DESCRIPTION
**Pull request by paulfantom**
Tests should be a bit faster with much simpler execution as there is no need to download multiple tools and setup minikube to work on travis.

This script also works outside of CI environment and can be used to run e2e tests locally.

/cc @brancz @LiliC @s-urbaniak

### CODE REVIEW

**Comment by s-urbaniak** [comment]
lgtm :tada:

**Comment by s-urbaniak** [comment]
feel free to self-merge, once green.

**Comment by paulfantom** [comment]
I added a loop to wait for the control plane to be up and squashed commits.

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
     # Run unit tests.
   - script: make test-unit
     # Run e2e tests.
-  - script: ./scripts/travis-e2e.sh
```
Comment (LiliC) [inline .travis.yml]: Do we not want to remove the old file?

Comment (paulfantom) [reply .travis.yml]: No, the consensus was that we want to save everything related to `minikube` until `kind` starts automatically publishing images.

**Comment by paulfantom** [comment]
Added conditional installation of `kubectl` as it is needed for the mechanism responsible for waiting for control plane.

**Comment by brancz** [comment]
Just curious, there seem to be a number of issues related to kind right now. Maybe it's not worth pursuing at this point and rather return to it at a later point?

**Comment by paulfantom** [comment]
Yes, let's put it on hold.

**Comment by paulfantom** [comment]
It looks like `TestAllNS/x/AMClusterAfterRollingUpdate` doesn't like to be run in kind :(

**Comment by paulfantom** [comment]
Verified locally that extending timeout helps in case of `AMClusterAfterRollingUpdate` test

**Comment by LiliC** [comment]
Alertmanager test failed, restarted the job.

**Comment by paulfantom** [comment]
Done in #3475

**Pull request closed without merging**